src/simulations/results/results_plotting_utils.jl

```diff
@@ -0,0 +1,406 @@
+    println("Threshold value for $(title_mapping[string(variables[1])][1]) when calculating statistics: $variable_title = $threshold")
+
+    # If "average" or "std" is in metrics, calculate them
+    if "average" in metrics || "std" in metrics
+        avg_vals = [mean(filter(x -> !(isnan(x)) && x >= threshold, matrix[:])) for matrix in data]
+        std_vals = [std(filter(x -> !(isnan(x)) && x >= threshold, matrix[:])) for matrix in data]
+    end
+
+    # Calculate and plot metrics
+    for metric in metrics
+        if metric == "average"
+            if "std" in metrics
+                band!(ax, t, avg_vals .- std_vals, avg_vals .+ std_vals, fillalpha=0.1, label="Std Dev", color=:lightgray)
+            end
+            lines!(ax, t, avg_vals, label="Average")
+        elseif metric == "median"
+            median_vals = [median(filter(x -> !isnan(x) && x >= threshold, matrix[:])) for matrix in data]
+            lines!(ax, t, median_vals, label="Median")
+        elseif metric == "min"
+            min_vals = [minimum(filter(x -> !isnan(x) && x >= threshold, matrix[:])) for matrix in data]
+            lines!(ax, t, min_vals, linestyle=:dot, label="Min")
+        elseif metric == "max"
+            max_vals = [maximum(filter(x -> !isnan(x) && x >= threshold, matrix[:])) for matrix in data]
+            lines!(ax, t, max_vals, linestyle=:dot, label="Max")
+        end
+        
+    end
+    leg = Legend(fig, ax)  
+    fig[1, 2] = leg 
+    
+    fig  # Return the main figure
+end
+
+"""
+    plot_glacier_IV(
+        results::T, 
+        variables::Vector{Symbol}; 
+        tspan::Tuple{Float64, Float64}, 
+        area::Float64 = 1.0
+    ) -> Figure
+
+Plot the temporal evolution of the integrated ice volume for a specific glacier variable.
+
+# Arguments
+- `results`: A custom type containing the results of a glacier simulation.
+  * Must have fields corresponding to the variables to be plotted.
+- `variables`: A vector of symbols representing the variable to be integrated and plotted. Only one variable is allowed.
+  * Possible variables: `:H` or any other data that has time evolution in it.
+- `tspan`: A tuple representing the start and end time for the simulation.
+- `area`: A float specifying the area (in square meters) of each pixel. This is used to calculate the integrated volume. Default is `1.0`.
+
+# Returns
+- A `Figure` object containing the plot showing the temporal evolution of integrated ice volume.
+
+# Notes
+- The function uses predefined titles for each variable based on a mapping, any other variable just get their symbol as title.
+- Only one variable can be plotted at a time with this function.
+- The integrated ice volume is calculated by summing up the values of the variable for each time step and multiplying by the provided pixel area.
+"""
+function plot_glacier_IV(results::T, variables::Vector{Symbol}; tspan::Tuple{Float64, Float64}, area::Float64 = 1.0)::Figure where T
```

Comment:
This function name is really obscure. I would find a better name that is easily understandable.